nibabel/nifti1.py

```diff
@@ -381,13 +381,89 @@ def write_to(self, fileobj, byteswap):
         fileobj.write(b'\x00' * (extstart + rawsize - fileobj.tell()))
 
 
+class Nifti1DicomExtension(Nifti1Extension):
+    """NIfTI1 DICOM header extension.
+
+    This class is a thin wrapper around pydicom to read a binary DICOM
+    byte string. If pydicom is not available, it silently falls back to the
+    standard NiftiExtension class.
+    """
+    def __init__(self, code, content, parent_hdr=None):
+        self._code = code
+        if parent_hdr:
+            self._is_little_endian = parent_hdr.endianness == '<'
+        else:
+            self._is_little_endian = True
+        if content.__class__ == Dataset:
+            self._is_implicit_VR = False
+            self._raw_content = self._mangle(content)
+            self._content = content
+        elif len(content):  # Got a byte string - unmangle it
```

Comment:
Yes, I think raising an error if it's neither of those - unless you want to allow None, to mean an empty dataset?

Yes, sorry, please wait for a bit - there were some little problems, will get back to you in 20 minutes or so.